cloud/amazon/ec2_vpc_vgw.py

```diff
@@ -0,0 +1,560 @@
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg=str(e))
+
+    status_achieved, vgw = wait_for_status(client, module, [vpn_gateway_id], 'attached')
+    if not status_achieved:
+        module.fail_json(msg='Error waiting for vpc to attach to vgw - please check the AWS console')
+
+    result = response
+    return result
+
+
+def detach_vgw(client, module, vpn_gateway_id, vpc_id=None):
+    params = dict()
+    params['VpcId'] = module.params.get('vpc_id')
+
+    if vpc_id:
+        try:
+            response = client.detach_vpn_gateway(VpnGatewayId=vpn_gateway_id, VpcId=vpc_id)
+        except botocore.exceptions.ClientError as e:
+            module.fail_json(msg=str(e))
+    else:
+        try:
+            response = client.detach_vpn_gateway(VpnGatewayId=vpn_gateway_id, VpcId=params['VpcId'])
+        except botocore.exceptions.ClientError as e:
+            module.fail_json(msg=str(e))
+
+    status_achieved, vgw = wait_for_status(client, module, [vpn_gateway_id], 'detached')
+    if not status_achieved:
+        module.fail_json(msg='Error waiting for  vpc to detach from vgw - please check the AWS console')
+
+    result = response
+    return result
+
+
+def create_vgw(client, module):
+    params = dict()
+    params['Type'] = module.params.get('type')
+
+    try:
+        response = client.create_vpn_gateway(Type=params['Type'])
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg=str(e))
+
+    result = response
+    return result
+
+
+def delete_vgw(client, module, vpn_gateway_id):
+
+    try:
+        response = client.delete_vpn_gateway(VpnGatewayId=vpn_gateway_id)
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg=str(e))
+
+    #return the deleted VpnGatewayId as this is not included in the above response
+    result = vpn_gateway_id
+    return result
+
+
+def create_tags(client, module, vpn_gateway_id):
+    params = dict()
+
+    try:
+        response = client.create_tags(Resources=[vpn_gateway_id],Tags=load_tags(module))
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg=str(e))
+
+    result = response
+    return result
+
+
+def delete_tags(client, module, vpn_gateway_id, tags_to_delete=None):
+    params = dict()
+
+    if tags_to_delete:
+        try:
+            response = client.delete_tags(Resources=[vpn_gateway_id], Tags=tags_to_delete)
+        except botocore.exceptions.ClientError as e:
+            module.fail_json(msg=str(e))
+    else:
+        try:
+            response = client.delete_tags(Resources=[vpn_gateway_id])
+        except botocore.exceptions.ClientError as e:
+            module.fail_json(msg=str(e))
+
+    result = response
+    return result
+
+
+def load_tags(module):
+    tags = []
+
+    if module.params.get('tags'):
+        for name, value in module.params.get('tags').iteritems():
+            tags.append({'Key': name, 'Value': str(value)})
```

Comment:
When defining a key with an empty value, this will fail:

```
AttributeError: 'str' object has no attribute 'iteritems'
```

It should be possible to define a tag key without value.